drift.py

```diff
@@ -35,31 +36,29 @@ def analyze(self, commit):
                 # If concretization successful check the resulting concrete specs.
                 if out.returncode == SUCCESS:
                     concrete_spec = spack.spec.Spec().from_yaml(out.stdout)
-                    if abstract_spec in self.last and self.last[abstract_spec] != concrete_spec:
-                        # Construct Result
-                        diff = spack.cmd.diff.compare_specs(self.last[abstract_spec], concrete_spec, colorful=False)
-                        result = Result(
-                            abstract_spec[:verToken],
-                            abstract_spec[verToken+1:],
-                            commit.hash,
-                            # Waiting for spack diff to complete tags.
-                            diff['b_not_a'],
-                            str(commit.author_date))
-                        # Send result to drift-server
-                        send(result)
-                    # Save concrete spec as last spec
+                    # If the spec is the same move on.
+                    if abstract_spec not in self.last or self.last[abstract_spec] == concrete_spec:
+                        continue
+                    # If the spec is different check what is different and save as tags.
+                    diff = spack.cmd.diff.compare_specs(
+                        self.last[abstract_spec],
+                        concrete_spec,
+                        colorful=False)
+                    tags = diff['b_not_a']
```

Comment:
So I think this line is going to need a little more parsing - this is a list of lists, so we would either want to morph the entries into one thing, or just take one of them. E.g., let's say we have this list:

```python
# an example
diff = {}
diff['b_not_a'] =  [
  [
   "version",
   "libbsd Version(0.11.3)"
  ],
  [
   "variant_value",
   "openssl docs bool(False)"
  ],
  [
   "node_compiler_version",
   "libbsd gcc Version(9.3.0)"
  ]
]
```
I think we'd want something like:

```python
tags = ["%s:%s" %(x[0],x[1]) for x in diff['b_not_a']]

['version:libbsd Version(0.11.3)',
 'variant_value:openssl docs bool(False)',
 'node_compiler_version:libbsd gcc Version(9.3.0)']
```
I was thinking of just including x[1], but I don't think the context would be sufficient for that.

Reply:
Good idea! I like separating the tags from their context by using a colon.